haystack/utils/doc_store.py

```diff
@@ -82,45 +73,60 @@ def launch_weaviate(sleep=15):
         time.sleep(sleep)
 
 
-def stop_opensearch():
-    logger.debug("Stopping OpenSearch...")
-    status = subprocess.run(["docker stop opensearch"], shell=True)
+def stop_container(container_name, delete_container=False):
+    logger.debug(f"Stopping {container_name}...")
+    status = subprocess.run([f"docker stop {container_name}"], shell=True)
     if status.returncode:
         logger.warning(
-            "Tried to stop OpenSearch but this failed. "
-            "It is likely that there was no OpenSearch Docker container with the name opensearch"
+            f"Tried to stop {container_name} but this failed. "
+            f"It is likely that there was no Docker container with the name {container_name}"
         )
-    status = subprocess.run(["docker rm opensearch"], shell=True)
+    if delete_container:
+        status = subprocess.run([f"docker rm {container_name}"], shell=True)
+
+
+def stop_opensearch(delete_container=False):
+    stop_container(OPENSEARCH_CONTAINER_NAME, delete_container)
+
+
+def stop_elasticsearch(delete_container=False):
+    stop_container(ELASTICSEARCH_CONTAINER_NAME, delete_container)
 
 
-def stop_service(document_store):
+def stop_milvus(delete_container=False):
+    stop_container(MILVUS1_CONTAINER_NAME, delete_container)
+
+
+def stop_weaviate(delete_container=False):
+    stop_container(WEAVIATE_CONTAINER_NAME, delete_container)
+
+
+def stop_service(document_store, delete_container=False):
     ds_class = str(type(document_store))
     if "OpenSearchDocumentStore" in ds_class:
-        stop_opensearch()
+        stop_opensearch(delete_container)
+    elif "ElasticsearchDocumentStore" in ds_class:
+        stop_elasticsearch(delete_container)
+    elif "MilvusDocumentStore" in ds_class:
+        stop_milvus(delete_container)
+    elif "WeaviateDocumentStore" in ds_class:
+        stop_weaviate(delete_container)
```

Comment:
If I understand correctly, `stop_weaviate` is nowhere else used except here. Could you please explain why we don't call `stop_container(WEAVIATE_CONTAINER_NAME, delete_container)`here directly and get rid of the `stop_weaviate` method?

Reply:
If someone's using weaviate, she should have the same launching/stopping functions as for the other dedicated-server-based doc_stores. So I think this is what a user would expect:
```python
from haystack.utils import launch_weaviate, stop_weaviate

launch_weaviate()

// do whatever with weaviate

stop_weaviate()
```